src/main/cpp/audioengine.cpp

```diff
@@ -232,18 +234,21 @@ namespace AudioEngine
         if ( recordFromDevice && AudioEngineProps::INPUT_CHANNELS > 0 )
         {
             int recSamps                  = DriverAdapter::getInput( recbufferIn );
-            SAMPLE_TYPE* recBufferChannel = recbuffer->getBufferForChannel( 0 );
+            SAMPLE_TYPE* recBufferChannel = inputChannel->getOutputBuffer()->getBufferForChannel( 0 );
 
             for ( int j = 0; j < recSamps; ++j )
-            {
                 recBufferChannel[ j ] = recbufferIn[ j ];//static_cast<float>( recbufferIn[ j ] );
 
-                // merge recording into current input buffer for instant monitoring
-                if ( monitorRecording )
-                {
-                    for ( int k = 0; k < outputChannels; ++k )
-                        inBuffer->getBufferForChannel( k )[ j ] = recBufferChannel[ j ];
-                }
+            // apply processing chain onto the input
+
+            std::vector<BaseProcessor*> processors = inputChannel->processingChain->getActiveProcessors();
+            for ( int k = 0; k < processors.size(); ++k )
+                processors[ k ]->process( inputChannel->getOutputBuffer(), AudioEngineProps::INPUT_CHANNELS == 1 );
```

Comment:
Now the input samples are written into an AudioChannel(s buffer), we can also enjoy the application of effects of the channels ProcessingChain.